code/game/atoms.dm

```diff
@@ -1378,16 +1378,30 @@
 	StartProcessingAtom(user, I, choices_to_options[pick])
 
 
-/atom/proc/StartProcessingAtom(mob/living/user, obj/item/I, list/chosen_option)
+/atom/proc/StartProcessingAtom(mob/living/user, obj/item/process_item, list/chosen_option)
+	var/processing_time = chosen_option[TOOL_PROCESSING_TIME]
 	to_chat(user, "<span class='notice'>You start working on [src]</span>")
-	if(I.use_tool(src, user, chosen_option[TOOL_PROCESSING_TIME], volume=50))
+	if(process_item.use_tool(src, user, processing_time, volume=50))
 		var/atom/atom_to_create = chosen_option[TOOL_PROCESSING_RESULT]
-		for(var/i = 1 to chosen_option[TOOL_PROCESSING_AMOUNT])
-			new atom_to_create(loc)
-		to_chat(user, "<span class='notice'>You manage to create [chosen_option[TOOL_PROCESSING_AMOUNT]] [initial(atom_to_create.name)] from [src]</span>")
-		qdel(src)
+		//var/list/atom/created_atoms = list() //Customfood
+		var/amount_to_create = chosen_option[TOOL_PROCESSING_AMOUNT]
+		for(var/i = 1 to amount_to_create)
+			var/atom/created_atom = new atom_to_create(drop_location())
+			created_atom.pixel_x = pixel_x
+			created_atom.pixel_y = pixel_y
+			if(i > 1)
+				created_atom.pixel_x += rand(-8,8)
+				created_atom.pixel_y += rand(-8,8)
```

Comment:
can we make a proc to do pixel offsets like this? i think its a common code pattern

Reply:
sure, ill go with a name like `randomize_placement`

Reply:
![image](https://github.com/BeeStation/BeeStation-Hornet/assets/62388554/e4ef2220-2b7c-4dd4-a5bb-c2f439d3795d)

procced in 9e80b25

Reply:
this doesnt need to be a proc on /atom/ since you aren't using src

Reply:
Yeah, I cant wrap my head around this, and it keeps throwing type info alerts at me however I try and spin it.

Pixels are randomized all over, and Its out of scope when it only accounts for two lines of code here